### ISSUE TITLE
Error with verbatim text in literalinclude caption Sphinx 1.4.4

### ISSUE THREAD

**Issue by andreacassioli** [sphinx-doc/sphinx#2676]
I have the following

```
   .. literalinclude:: unicode.c
      :caption: An example how to convert a ``wchar_t`` string to a UTF8 string.
      :name: doc.limit.utf8
```

If I generate Tex and build a PDF using pdflatex I get

```
! Undefined control sequence.
\@noligs ...lig@list \let '\textquotesingle \let `
                                                  \textasciigrave \ifx \enco...
l.2546 \begin{Verbatim}[commandchars=\\\{\}]
```

That basically complaints about 

```
\def\SphinxLiteralBlockLabel{\label{limitation:doc-limit-utf8}}
\SphinxSetupCaptionForVerbatim{literal-block}{An example how to convert a \code{wchar\_t} string to a UTF8 string.}
\begin{Verbatim}[commandchars=\\\{\}]
```

It seems that removing the verbatim string from the caption solves the issue.

This worked fine in 1.4.3.

**Comment by jfbu** [comment]
oh sorry, that's my fault with <strike>bad interaction of</strike> the recent change at 82758b1 <strike>with the set-up to break long code lines. The set-up is already in place at the time of the caption, hence the problem.</strike> Will fix.

**Comment by jfbu** [comment]
actually, although indeed the cause is the change at 82758b1, my initial diagnostic was wrong, it is a general issue of inline code inside captions, not only the ones for literalinclude or code-block.

```
.. figure:: pic.png

   ``wchar_t``
```

Fix in the way.

**Comment by jfbu** [comment]
The cause was in PR#2627 addition to `\code` of the "fragile" macro `\@noligs` from the LaTeX kernel. Fixed at      8608387. Does this work for you ?
